src/types/markers.js

```diff
@@ -19,14 +19,19 @@ export type CauseBacktrace = {|
 |};
 
 /**
- * This type holds data that should be synchronized between the sender and
- * recipient sides of an IPC marker.
+ * This type holds data that should be synchronized across the various phases
+ * associated with an IPC message.
  */
-export type IPCPairData = {|
-  startTime: Milliseconds,
-  endTime: Milliseconds,
-  otherTid: number,
-  otherThreadName: string,
+export type IPCSharedData = {|
+  startTime?: Milliseconds,
```

Comment:
Please add a comment here explaining why these types can be optional. Also, this type is a bit confusing between the _derived_ payload and _Gecko_ payload. It would be nice to add `_Gecko` types. See in this file for other examples, and then add comments about how the information gets derived.

Reply:
Done.